internal/yarpcerrors/yarpcerrors.go

```diff
@@ -31,3 +35,16 @@ import "go.uber.org/yarpc/yarpcerrors"
 func NewWithNamef(code yarpcerrors.Code, name string, format string, args ...interface{}) *yarpcerrors.Status {
 	return yarpcerrors.Newf(code, format, args...).WithName(name)
 }
+
+// AnnotateWithInfo will take an error and add info to it's error message while
+// keeping the same status code.
+func AnnotateWithInfo(status *yarpcerrors.Status, format string, args ...interface{}) *yarpcerrors.Status {
+	return yarpcerrors.Newf(status.Code(), "%s: %s", sprintf(format, args...), status.Message())
```

Comment:
Should the format, args come after the message? Alternatively I might just force people to do a string instead of args... because this could get weird

Reply:
yeah, not really sure, depends on what people are doing, since this is internal this should probably be fine for now.